.travis.yml

```diff
@@ -5,23 +5,27 @@ lein: 2.7.1
 # based on when the repository testing was enabled.
 sudo: required
 
-jdk:
-  - openjdk7
-  - oraclejdk8
-  - oraclejdk7
-
-env:
-  - PDB_TEST_LANG=clojure
-  - PDB_TEST_LANG=ruby
-
-# Only run one ruby test
 matrix:
-  exclude:
-    - jdk: oraclejdk8
-      env: PDB_TEST_LANG=ruby
-    - jdk: oraclejdk7
-      env: PDB_TEST_LANG=ruby
-
+  include:
+    # terminus unit tests (spec tests)
+    - env: PDB_TEST_LANG=ruby PUPPET_VERSION=latest
+    - env: PDB_TEST_LANG=ruby PUPPET_VERSION=4.0.0
+
+    # pdb unit tests
+    - env: PDB_TEST_LANG=clojure PDB_TEST_SELECTOR=:unit
+      jdk: openjdk7
+    - env: PDB_TEST_LANG=clojure PDB_TEST_SELECTOR=:unit
+      jdk: oraclejdk7
+    - env: PDB_TEST_LANG=clojure PDB_TEST_SELECTOR=:unit
+      jdk: oraclejdk8
+
+    # pdb + terminus + puppetserver + puppet integration tests
+    - env: PDB_TEST_LANG=clojure PDB_TEST_SELECTOR=:integration PUPPET_VERSION=4.6.2 PUPPETSERVER_VERSION=2.7.2
+      jdk: oraclejdk8
+    - env: PDB_TEST_LANG=clojure PDB_TEST_SELECTOR=:integration PUPPET_VERSION=latest PUPPETSERVER_VERSION=master
```

Comment:
Not critical - just thought it might be nice if we happened to know they supported that.